src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestActions/DeclineAction.js

```diff
@@ -0,0 +1,133 @@
+import { documentApi } from '@api/documents';
+import { ESSelectorModal } from '@modules/ESSelector';
+import { serializeDocument } from '@modules/ESSelector/serializer';
+import get from 'lodash/get';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Confirm, Dropdown } from 'semantic-ui-react';
+
+export class DeclineAction extends React.Component {
+  state = {
+    header: null,
+    type: null,
+    confirmModalOpened: false,
+  };
+
+  onCancel = () => {
+    this.setState({ confirmModalOpened: false });
+  };
+
+  onConfirm = (type) => {
+    const { onDecline } = this.props;
+    this.setState({ confirmModalOpened: false });
+    onDecline({ decline_reason: type });
+  };
+
+  onDeclineWithDocument = (selections) => {
+    const { onDecline } = this.props;
+
+    onDecline({
+      decline_reason: 'IN_CATALOG',
+      document_pid: get(selections, '0.id'),
+    });
+  };
+
+  onClick = (event, { text, value }) => {
+    if (['USER_CANCEL', 'NOT_FOUND', 'OTHER'].includes(value)) {
+      this.setState({
+        header: text,
+        type: value,
+        confirmModalOpened: true,
+      });
+    } else {
+      throw new Error(`Invalid decline type: ${value}`);
+    }
+  };
+
+  renderOptions() {
+    const options = [
+      {
```

Comment:
shouldn't this be a vocab ?

Reply:
No, it is not, it is an hardcoded enum. This is because the system is doing different things based on that value, it is affecting behavior.